python/test/pyElement_TEST.py

```diff
@@ -0,0 +1,479 @@
+                         parent.get_first_element().file_path())
+        self.assertEqual("1.5", parent.get_first_element().original_version())
+
+        parent.clear_elements()
+
+        self.assertEqual(parent.get_first_element(), None)
+        self.assertEqual("/path/to/file.sdf", parent.file_path())
+        self.assertEqual(12, parent.line_number())
+        self.assertEqual("1.5", parent.original_version())
+
+    def test_clear(self):
+        parent = Element()
+        child = Element()
+
+        parent.set_file_path("/path/to/file.sdf")
+        parent.set_line_number(12)
+        parent.set_xml_path("/sdf/world[@name=\"default\"]")
+        parent.set_original_version("1.5")
+        child.set_parent(parent)
+        parent.insert_element(child)
+
+        self.assertEqual("/path/to/file.sdf", parent.file_path())
+        self.assertEqual(12, parent.line_number())
+        self.assertEqual("/sdf/world[@name=\"default\"]", parent.xml_path())
+        self.assertEqual("1.5", parent.original_version())
+        self.assertNotEqual(parent.get_first_element(), None)
+        self.assertEqual("/path/to/file.sdf",
+                         parent.get_first_element().file_path())
+        self.assertEqual("1.5", parent.get_first_element().original_version())
+
+        parent.clear()
+
+        self.assertEqual(parent.get_first_element(), None)
+        self.assertEqual(parent.file_path(), "")
+        self.assertIsNone(parent.line_number())
+        self.assertEqual(parent.xml_path(), "")
+        self.assertEqual(parent.original_version(), "")
+
+    def test_to_string_elements(self):
+        parent = Element()
+        child = Element()
+
+        parent.set_name("parent")
+        child.set_name("child")
+
+        parent.insert_element(child)
+        self.assertNotEqual(parent.get_first_element(), None)
+
+        parent.add_attribute("test", "string", "foo", False, "foo description")
+        self.assertEqual(parent.get_attribute_count(), 1)
+
+        # attribute won't print unless it has been set
+        self.assertFalse(parent.get_attribute_set("test"))
+        self.assertEqual(
+            parent.to_string("<!-- prefix -->"), "<!-- prefix --><parent>\n"
+            "<!-- prefix -->  <child/>\n"
+            "<!-- prefix --></parent>\n")
+
+        test = parent.get_attribute("test")
+        self.assertNotEqual(None, test)
+        self.assertFalse(test.get_set())
+        self.assertTrue(test.set_from_string("foo"))
+        self.assertTrue(test.get_set())
+        self.assertTrue(parent.get_attribute_set("test"))
+
+        self.assertEqual(
+            parent.to_string("<!-- prefix -->"),
+            "<!-- prefix --><parent test='foo'>\n"
+            "<!-- prefix -->  <child/>\n"
+            "<!-- prefix --></parent>\n")
+
+    def test_set(self):
+        elem = Element()
+
+        elem.add_value("string", "val", False, "val description")
+
+        self.assertTrue(elem.set_string("hello"))
```

Comment:
Can we also get the value from `elem` and assert equality in order to check it is properly stored on the object?

Reply:
Done in https://github.com/gazebosim/sdformat/pull/1303/commits/780059283b4a76e1098a59b1ab83aa36876eb25c